src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/Grpc/GrpcProtocolHelpers.cs

```diff
@@ -29,7 +29,7 @@ internal static class GrpcProtocolHelpers
     internal const string MessageTrailer = "grpc-message";
     internal const string CancelledDetail = "No grpc-status found on response.";
 
-    public static Status? GetResponseStatus(HttpHeaders trailingHeaders, HttpResponseMessage httpResponse)
+    public static Status? GetResponseStatus(HttpResponseMessage httpResponse, HttpHeaders trailingHeaders)
```

Comment:
This code is a bit funny. It is defined to return `Nullable<Status>` yet seems to always return a status.

Suggest re-writing something like this:

```csharp
    public static Status GetResponseStatus(HttpResponseMessage httpResponse, HttpHeaders trailingHeaders)
    {
        try
        {
            return trailingHeaders.Any()
                ? GetStatusCore(trailingHeaders)
                : GetStatusCore(httpResponse.Headers);
        }
        catch (Exception ex)
        {
            // Handle error from parsing badly formed status
            return new Status(StatusCode.Cancelled, ex.Message, ex);
        }
    }

    public static Status GetStatusCore(HttpHeaders headers)
    {
        var grpcStatus = GetHeaderValue(headers, StatusTrailer);

        // grpc-status is a required trailer
        if (grpcStatus == null)
        {
            // return new Status(StatusCode.Cancelled, CancelledDetail); // Exisiting logic creates this
            return Status.CancelledNoReply; // Instead define a static because it seems to be reusable
        }

        int statusValue;
        if (!int.TryParse(grpcStatus, out statusValue))
        {
            throw new InvalidOperationException("Unexpected grpc-status value: " + grpcStatus);
        }

        // grpc-message is optional
        // Always read the gRPC message from the same headers collection as the status
        var grpcMessage = GetHeaderValue(headers, MessageTrailer);

        if (!string.IsNullOrEmpty(grpcMessage))
        {
            // https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses
            // The value portion of Status-Message is conceptually a Unicode string description of the error,
            // physically encoded as UTF-8 followed by percent-encoding.
            grpcMessage = Uri.UnescapeDataString(grpcMessage);
        }

        return new Status((StatusCode)statusValue, grpcMessage ?? string.Empty);
    }
```

That should help clean up callers who will no longer need to deal with the nullable/hasvalue check?

Reply:
Fixed